packages/core/src/lambda/vue/remoteInvoke/invokeLambda.ts

```diff
@@ -443,22 +436,164 @@ export class RemoteInvokeWebview extends VueWebview {
+        const options = events.length > 0 ? [createNewOption, ...events] : [createNewOption]
+
+        const selected = await vscode.window.showQuickPick(options, {
+            placeHolder: localize(
+                'AWS.lambda.remoteInvoke.saveEventChoice',
+                'Create new or overwrite existing test event'
+            ),
+            title: localize('AWS.lambda.remoteInvoke.saveRemoteEvent', 'Save as Remote Event'),
+        })
+
+        if (!selected) {
+            return undefined
+        }
+
+        let eventName: string | undefined
+
+        if (selected === createNewOption) {
+            // Prompt for new event name
+            eventName = await vscode.window.showInputBox({
+                prompt: localize('AWS.lambda.remoteInvoke.enterEventName', 'Enter a name for the test event'),
+                placeHolder: localize('AWS.lambda.remoteInvoke.eventNamePlaceholder', 'MyTestEvent'),
+                validateInput: (value) => {
+                    if (!value || value.trim() === '') {
+                        return localize('AWS.lambda.remoteInvoke.eventNameRequired', 'Event name is required')
+                    }
+                    if (events.includes(value)) {
+                        return localize(
+                            'AWS.lambda.remoteInvoke.eventNameExists',
+                            'An event with this name already exists'
+                        )
+                    }
+                    return undefined
+                },
+            })
+        } else {
+            // Use selected existing event name
+            const confirm = await showConfirmationMessage({
+                prompt: localize(
+                    'AWS.lambda.remoteInvoke.overwriteEvent',
+                    'Overwrite existing test event "{0}"?',
+                    selected
+                ),
+                confirm: localize('AWS.lambda.remoteInvoke.overwrite', 'Overwrite'),
+                cancel: 'Cancel',
+                type: 'warning',
+            })
+
+            if (confirm) {
+                eventName = selected
+            }
+        }
+
+        if (eventName) {
+            const eventData = {
+                name: eventName,
+                event: eventContent,
+                region: region,
+                arn: functionArn,
+            }
+            // Use force flag when overwriting existing events
+            const isOverwriting = selected !== createNewOption
+            await this.createRemoteTestEvents(eventData, isOverwriting)
+            return eventName
+        }
+
+        return undefined
+    }
+
+    public async createRemoteTestEvents(putEvent: Event, force: boolean = false) {
         const params: SamCliRemoteTestEventsParameters = {
             functionArn: putEvent.arn,
             operation: TestEventsOperation.Put,
             name: putEvent.name,
             eventSample: putEvent.event,
             region: putEvent.region,
+            force: force,
```

Comment:
I'd like to discuss this parameter's usage. Since it appears to be specific to handling test event overwrites, would it be cleaner to move this logic directly into `saveRemoteTest()` to keep the code clean?